python/cudnn/gemm_swiglu/api.py

```
@@ -104,43 +107,35 @@ def __init__(
-            rest_n2 = ceil_div(ceil_div(n // 2, self.sf_vec_size), 4)
-            self._check_tensor_shape(self.sfc_desc, (32, 4, ceil_div(m, 128), 4, rest_n2, l), "SFC")
-            self._check_tensor_shape(self.norm_const_desc, (1,), "norm_const")
-
-        _ = self._check_tensor_stride(self.a_desc, stride=[(1, m, m * k), (k, 1, m * k)])
-        _ = self._check_tensor_stride(self.b_desc, stride=[(1, n, n * k), (k, 1, n * k)])
-        _ = self._check_tensor_stride(self.ab12_desc, stride=[(1, m, m * n), (n, 1, m * n)])
-        _ = self._check_tensor_stride(self.c_desc, stride=[(1, m, m * n_2), (n_2, 1, m * n_2)])
-        self._value_error_if(
-            self.ab12_desc.stride_order != self.c_desc.stride_order,
-            f"AB12 and C tensor stride orders must match, got {self.ab12_desc.stride_order} and {self.c_desc.stride_order}",
+        m, n, k, l = require_gemm_shapes(
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)